### PR TITLE
Sqlserver support 2

### PR DESCRIPTION
**Pull request by rPraml**
This is the next PR that adds a lot of sqlserver fixes, but please merge the first one.

### CODE REVIEW

File: src/main/java/io/ebean/config/dbplatform/sqlserver/SqlServerPlatform.java [hunk 1]
```diff
@@ -19,23 +24,22 @@
...
     this.idInExpandedForm = true;
     this.selectCountWithAlias = true;
     this.sqlLimiter = new SqlServerSqlLimiter();
     this.basicSqlLimiter = new SqlServerBasicSqlLimiter();
     this.platformDdl = new SqlServerDdl(this);
     this.historySupport = new SqlServerHistorySupport();
-    this.dbIdentity.setIdType(IdType.IDENTITY);
-    this.dbIdentity.setSupportsGetGeneratedKeys(true);
-    this.dbIdentity.setSupportsIdentity(true);
+    dbIdentity.setIdType(IdType.SEQUENCE);
```
Comment (rbygrave) [inline src/main/java/io/ebean/config/dbplatform/sqlserver/SqlServerPlatform.java]: We can't just flip to sequence here without breaking people ... meaning we need a plan (like a ServerConfig option) to allow people to migrate.

Absolutely we want SqlServer folks to move to sequences so we do want this ... just need to be mindful of people already somewhat committed to IDENTITY.

**Comment by rbygrave** [comment]
This PR has overlap with the prior PR.  So I think this can't be merged in it's current state.

File: src/main/java/io/ebean/config/dbplatform/sqlserver/SqlServerPlatform.java (hunk 1 above)
Comment (rPraml) [inline src/main/java/io/ebean/config/dbplatform/sqlserver/SqlServerPlatform.java]: I Agree, but before introducing a complex migration path, we should ask who already uses sqlserver productive.
As there were so many bugs and also posts on the google groups I think no one uses it really productive, so I would suggest to do a hard switch from IDENTITY to SEQUENCE. For the few pepole this means either to recreate the database or write some custom migration ddls
If most people do not have a problem

Comment (rbygrave) [inline src/main/java/io/ebean/config/dbplatform/sqlserver/SqlServerPlatform.java]: Yes, perhaps the best option is to have a platform that extends SqlServerPlatform and overrides to use IDENTITY etc.

**Pull request closed without merging**